leshan-core/src/main/java/org/eclipse/leshan/senml/json/jackson/SenMLJsonRecordSerDes.java

```diff
@@ -105,7 +106,7 @@ else if (value instanceof Float) {
                 jsonObj.put("vlo", record.getObjectLinkValue());
                 break;
             case OPAQUE:
-                jsonObj.put("vd", Base64.encodeBase64String(record.getOpaqueValue()));
+                jsonObj.put("vd", new DefaultBase64Encoder(true, true).encode(record.getOpaqueValue()));
```

Comment:
Should we store the encoder in an attribute ?
(same for decoder ?)

Reply:
We actually discussed it in #1348 in [this comment](https://github.com/eclipse/leshan/issues/1348#issue-1447767894):

>And finally in our custom version of SenMLJsonRecordSerDes, we basically have 2 lines of code to change with respect to the Leshan version (related to OPAQUE fields encoding and decoding):

```java
case OPAQUE:
    jsonObj.put("vd", urlSafeEncoder.encode(record.getOpaqueValue()));
    break;
```
>and
```java
JsonNode vd = o.get("vd");
if (vd != null && vd.isTextual()) {
    record.setOpaqueValue(urlSafeDecoder.decode(vd.asText()));
    hasValue = true;
}
```

So that would require better customization of `SenMLJsonRecordSerDes` class and others.

Reply:
I'm not sure I get you :thinking: 

My point was just to avoid a decoder(or encoder) object creation by value to encode or decode.

I'm not sure this is related to better customization ? 

Or maybe you mean : "I will change this later anyway so we don't need to clean this now ?"

Reply:
I think it depends if we will call decoders/encoders with different options often or not. I probably would lean towards:
```java
jsonObj.put("vd", Base64.encodeBase64String(record.getOpaqueValue()));
```
But if we aim to add more options for  decoders/encoders or change those options often, we should probably stick to creating new decoders/encoders.

Reply:
We want to be able to let user creates a `SenMLJsonRecordSerDes` with a defirent base64 decoder than the one by default. (discussion at #1348)

But I don't think we want to be able to change `SenMLJsonRecordSerDes` behavior dynamically.

I mean once we create a `SenMLJsonRecordSerDes` with a given base64 decoder we don't change it.


So my idea was to create an attribute initialized with `new DefaultBase64Encoder(true, true)` then use it here.

Reply:
In that case i would initialize  decoder once and then reuse it